.gitattributes

```diff
@@ -1,7 +1,8 @@
 * text=auto eol=lf
+
 # git grep shouldn't match entries in this benchmark data
 bench_data/** binary
-crates/syntax/test_data/** -text eof=LF
```

Comment:
Why was this removed?

Reply:
It's a fun story! The short answer it: it caused bors to fail to merge the `rust-lang/rust` subtree PR.

The longer story is: that change was introduced in https://github.com/rust-lang/rust-analyzer/pull/2166 - but it never did what was intended.

`eof` is not a gitattribute setting. `eol` is, but even that line used `eol`, it _still_ wouldn't do anything, because of `-text`. So by mistake, 3 files were treated as binary, and checked out with CRLF line endings (those were the ones bors choked on).

After removing this line from `.gitattribute`, I re-added the folder to git, removed it from disk and used `git restore` to write it with the proper line endings. The relevant commit is here: https://github.com/rust-lang/rust/commit/1bb63df1fdd40cbb0f3b1c9d5a838f9195142a07

Reply:
I just did a fresh clone of that branch from Windows and ran `cargo test` with the latest stable, no issues whatsoever.

I'm still convinced #2166 wasn't needed, as explained here: https://github.com/rust-lang/rust-analyzer/pull/2166#discussion_r928246345